client/src/HoverProvider.ts

```diff
@@ -0,0 +1,179 @@
+'use strict';
+import * as vscode from 'vscode';
+import { CoqProject } from './CoqProject';
+import * as editorAssist from './EditorAssist';
+
+// Inputs to ignore when calling hover
+const excludes = [''];
+
+function operatorRegex(str: string) {
+  // Matching operators is simple, as Coq will kindly
+  // print spaces before and after them,
+  // we use lookahead/lookbehind to avoid catching them
+  return new RegExp("(?<=\\s)" + str + "(?=\\s)", "g");
+}
+
+// Format a Coq type to be pretty and compact (e.g. replace forall with ∀)
+function compactify(str: string) {
+  const replaces = [
+    { match: /\bfun\b/g, subst: "λ" },
+    { match: /\bforall\b/g, subst: "∀" },
+    { match: /\bexists\b/g, subst: "∃" },
+    { match: operatorRegex("\\\\\\/"), subst: "∨" },
+    { match: operatorRegex("\\/\\\\"), subst: "∧" },
+    { match: operatorRegex("<->"), subst: "↔" },
+    { match: operatorRegex("->"), subst: "→" },
+    { match: operatorRegex("<="), subst: "≤" },
+    { match: operatorRegex(">="), subst: "≥" },
+    { match: operatorRegex("=>"), subst: "⇒" },
+    { match: operatorRegex("<>"), subst: "≠" },
+    { match: operatorRegex("~"), subst: "¬" }
+  ];
+  for (const replace of replaces) {
+    str = str.replace(replace.match, replace.subst);
+  }
+  return str.trim();
+}
+
+function formatCheck(response: string) {
+  // response is the string printed by "Check a." :
+  // |a
+  // |\t : Type
+  // |       type (continued if long) (7 space indent)
+  // |where
+  // |?optional = whatever
+
+  // § Strip output of anything but the type
+  const array = response.split("\nwhere\n");
+  let type = array[0];
+  // let where = array[1];
+  type = type.replace(/^.*?\n\t : /, ""); // remove identifier
+  type = type.replace(/^ {7}/gm, ""); // remove indent
+
+  if (type === "") return;
+  type = compactify(type);
+
+  let hover = [{ language: "coq", value: type }];
+  // if (where)
+  //  hover.push({language:"coq", value:where});
+  return new vscode.Hover(hover);
+}
```

Comment:
@dlesbre How hard w you replace `Check function.` by `About function.`? If `function` takes implicit arguments, `Check` can be somewhat unhelpful:

![image](https://user-images.githubusercontent.com/289960/207205788-4d178e1a-a4c9-48f8-872d-89e6b3085f87.png)

```coq
About set_map.
(*
set_map :
∀ {A C : Type},
  Elements A C
  → ∀ {B D : Type}, Singleton B D → Empty D → Union D → (A → B) → C → D

set_map is not universe polymorphic
Arguments set_map {A C}%type_scope {H} {B D}%type_scope 
  {H0 H1 H2} f%function_scope X
set_map is transparent
Expands to: Constant stdpp.fin_sets.set_map
*)
Check set_map.
(*
set_map
	 : (?A → ?B) → ?C → ?D
where
?A : [ |- Type]
?C : [ |- Type]
?H : [ |- Elements ?A ?C]
?B : [ |- Type]
?D : [ |- Type]
?H0 : [ |- Singleton ?B ?D]
?H1 : [ |- Empty ?D]
?H2 : [ |- Union ?D]
*)
```

Reply:
Replacing one with the other is fairly simple. I used Check more or less as a default since it's easy to parse if you just want the quick and short type.

I was concerned about hovers becoming too long (which is also why I cut and format the output of Check to keep it minimal).

I'll try using 'About' in a few hours

Reply:
Maybe it could be a user setting to pick which? (Unless it's too much work of course.)

Reply:
I added 'About' and it seems to work fine. I made the choice between Check and About a user setting that default to "about".

The hover using about also provides additional information (namely, the "Arguments" line which shows the scope and implicit status of arguments, and the "Expands to: " line)